### ISSUE TITLE
144hz over USB-C DP alt mode

### ISSUE THREAD

**Issue by qdrop17** [jrandiny/yoga-slim7-ubuntu#44]
Hi everyone.

Does anyone of you manage to push 144hz over USB-C DP alt mode (I'm using a HP Dock G5 with an ultrawide 3440x1440 screen)?

I'm facing issues with the amdgpu driver I guess:
[https://gitlab.freedesktop.org/drm/amd/-/issues/1616](https://gitlab.freedesktop.org/drm/amd/-/issues/1616)

This is the only feature, that keeps me from using Linux on my Slim-7 as a daily driver. I have a gaming PC that I'm using with gamestreaming. But I want to push 144 frames per second.

This is working flawlessly under Windows...

Any help is highly appreciated.

Best

qdrop

**Comment by gregorkrz** [comment]
@qdrop17 , have you had any issues with DP over USB-C at 60 Hz or less? I have observed weird behaviour: when launching some apps (e. g. `vscode` or `virtualbox`) or importing specific Python libraries (`gensim`, `matpotlib`) the connection simply dies for a couple of seconds - in dmesg it looks just like one would expect when unplugging the port and plugging it back. Sometimes, it keeps repeating that pattern and have to stop it by unplugging the port. No issues with the same display over HDMI. Kernel 5.8.10.

**Comment by qdrop17** [comment]
> 
> 
> @qdrop17 , have you had any issues with DP over USB-C at 60 Hz or less? I have observed weird behaviour: when launching some apps (e. g. `vscode` or `virtualbox`) or importing specific Python libraries (`gensim`, `matpotlib`) the connection simply dies for a couple of seconds - in dmesg it looks just like one would expect when unplugging the port and plugging it back. Sometimes, it keeps repeating that pattern and have to stop it by unplugging the port. No issues with the same display over HDMI. Kernel 5.8.10.
> 
> EDIT:
> Running `echo 0x104 > /sys/module/drm/parameters/debug` does seem to reveal slightly more info in dmesg. Here is the output:
> [dmesg_140721_0x104.txt](https://github.com/jrandiny/yoga-slim7-ubuntu/files/6814290/dmesg_140721_0x104.txt)

no, I never experienced such issues. Sounds like a bad cable or something?

**Comment by gregorkrz** [comment]
Nope, the same on different cables, even different brands of displays.

**Comment by qdrop17** [comment]
I added a comment here: https://gitlab.freedesktop.org/drm/amd/-/issues/1616

I think it has to do with how the USB-C lanes are split; for some reason, Linux splits these differently than windows. Keywords: Display Port 1.4 (HBR3 / DSC).
-> When using a USB-C to DP adapter it works as intended (as there is more than enough bandwith for just one DP port I suppose).
-> When using a USB-C Dock (with 2x DP, USB, Network, Audio, etc...) it does not work with higher refreshrates (3440x1440 above 60hz).
How can we take control over how this USB-C bandwith is allocated? I would LOVE to switch to Linux as a daily driver (also for gaming) - but not being able to run 144hz is no option for me.
Windows manages to easily push gigabit over the ethernetport as well as 3440x1440 at 144hz at the same time.
Any help is highly appreciated.
